include/seastar/core/sstring.hh

```diff
@@ -883,7 +883,7 @@ std::ostream& operator<<(std::ostream& os, const std::unordered_map<Key, T, Hash
 SEASTAR_MODULE_EXPORT
 template <typename char_type, typename Size, Size max_size, bool NulTerminate>
 struct fmt::formatter<seastar::basic_sstring<char_type, Size, max_size, NulTerminate>>
-    : private fmt::formatter<std::basic_string_view<char_type>> {
+    : public fmt::formatter<std::basic_string_view<char_type>> {
```

Comment:
the reason why i chose to use `private` inheritance was to avoid the disputations based on Liskov principal (the IS-A principal). but if we need to workaround this compiler bug. i think i'd give up and just use the public inheritance, as it's simpler:

1. drop `private`.
2. drop `using base::parse`
3. optionally replace `base` with `fmt::formatter<format_as_t>`. as it is not reused.

Reply:
I agree that private is better here, I switched to public to make it work, we can add a comment referencing the clang bug once we know what it is.